doc/architecture-specification.md

```diff
@@ -108,19 +110,21 @@ These keywords `SHALL` be in `monospace` for ease of identification.
 
 ### Trading
 
-1. Orders `SHALL` accept the object address of a user's market account, as well
-   as a size, limit price (worst execution price including fees), and
-   restriction.
+1. A user's open orders `SHALL` be tracked in an open orders object, such that
+   market makers can submit the address of the object for optimized order
+   placement.
+1. Orders `SHALL` accept a size, limit price (worst execution price including
+   fees), restriction, and open orders object address.
+1. Order APIs `SHALL` accept `0x0` for open orders address to indicate an
+   immediate-or-cancel order that settles to a user's primary fungible store.
 1. Limit orders, market orders, and swaps `SHALL` be moderated via a single API
-   that uses `HI_64` or `0` to indicate an immediate-or-cancel market order.
-1. Orders `SHALL` fill outside of a market account if the market account object
-   address is passed as `0x0`.
+   that uses `HI_64` or `0` for limit price to indicate an immediate-or-cancel
+   market order.
```

Comment:
```suggestion
   wherein a user can set an order's limit price to `HI_64` or `0` to indicate an immediate-or-cancel
   market order.
```

I think this is could be a bit clearer if I'm understanding correctly.

Am I right in reading this as a user can effectively create a market order by setting a limit order's limit price to `HI_64` or `0`?

Reply:
Yes this is correct, and nice catch here: this should actually be updated to reflect the new rational number pricing paradigm, as I've done in 3aab499